pkg/tkctl/cmd/upinfo/upinfo.go

```diff
@@ -0,0 +1,206 @@
+
+	clientConfig, err := tkcContext.ToTkcClientConfig()
+	if err != nil {
+		return err
+	}
+
+	if tidbClusterName, ok := clientConfig.TidbClusterName(); ok {
+		o.TidbClusterName = tidbClusterName
+	} else {
+		return cmdutil.UsageErrorf(cmd, infoUsage)
+	}
+
+	namespace, _, err := clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+	o.Namespace = namespace
+
+	restConfig, err := clientConfig.RestConfig()
+	if err != nil {
+		return err
+	}
+	tcCli, err := versioned.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	o.TcCli = tcCli
+	kubeCli, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	o.KubeCli = kubeCli
+
+	return nil
+}
+
+func (o *UpInfoOptions) Run() error {
+
+	tc, err := o.TcCli.PingcapV1alpha1().
+		TidbClusters(o.Namespace).
+		Get(o.TidbClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	setName := controller.TiDBMemberName(tc.Name)
+	set, err := o.KubeCli.AppsV1beta1().StatefulSets(o.Namespace).Get(setName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	podList, err := o.KubeCli.CoreV1().Pods(o.Namespace).List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s,%s=%s", label.InstanceLabelKey, tc.Name, label.ComponentLabelKey, "tidb"),
+	})
+	if err != nil {
+		return err
+	}
+	msg, err := renderTCUpgradeInfo(tc, set, podList)
+	if err != nil {
+		return err
+	}
+	fmt.Fprint(o.Out, msg)
+	return nil
+}
+
+func renderTCUpgradeInfo(tc *v1alpha1.TidbCluster, set *apps.StatefulSet, podList *v1.PodList) (string, error) {
+	return readable.TabbedString(func(out io.Writer) error {
+		w := readable.NewPrefixWriter(out)
+		dbPhase := tc.Status.TiDB.Phase
+		w.WriteLine(readable.LEVEL_0, "Name:\t%s", tc.Name)
+		w.WriteLine(readable.LEVEL_0, "Namespace:\t%s", tc.Namespace)
+		w.WriteLine(readable.LEVEL_0, "CreationTimestamp:\t%s", tc.CreationTimestamp)
+		w.WriteLine(readable.LEVEL_0, "Statu:\t%s", dbPhase)
+		{
+			w.WriteLine(readable.LEVEL_1, "Name\tState\t")
+			w.WriteLine(readable.LEVEL_1, "----\t-----\t")
+			{
+				updateReplicas := set.Spec.UpdateStrategy.RollingUpdate.Partition
+
+				if len(podList.Items) != 0 {
+					for _, pod := range podList.Items {
+						var state string
+						ordinal, err := util.GetOrdinalFromPodName(pod.Name)
+						if err != nil {
+							return err
+						}
+						if dbPhase == v1alpha1.UpgradePhase {
+							if (*updateReplicas) < ordinal {
+								state = UPDATED
+							} else if (*updateReplicas) == ordinal {
+
+								state = UPDATING
+
+								if pod.Labels[apps.ControllerRevisionHashLabelKey] == tc.Status.TiDB.StatefulSet.UpdateRevision {
+									if member, exist := tc.Status.TiDB.Members[pod.Name]; exist && member.Health {
+										state = UPDATED
+									}
+								}
+
+							} else {
+								state = WAITING
+							}
+						} else {
+							state = UPDATED
+						}
+						w.WriteLine(readable.LEVEL_1, "%s\t%s\t", pod.Name, state)
```

Comment:
Do we need to display the image before and after the upgrade?

Reply:
It seems reasonable

Reply:
What is the display effect?

Reply:
show the output of the upinfo command like aylei does: https://github.com/pingcap/tidb-operator/pull/387#issue-269955716

Reply:
I've already show it in <What is changed and how it works?> and I bold it.